### DISCUSSION TITLE
Weird bug when using two screens. Programs on second screen switch while working in xournal

### DISCUSSION THREAD

**Discussion by KonstantinHebenstreit** [xournalpp/xournalpp#2958] [Q&A]
Hey! Just started using the nightly build and ran into some very weird behavior. I hope someone can help me. I do not even know for sure if it is a bug of Xournal or of Ubuntu.

I think it could be a bug of Xournal, since it never accurs when I used the stable version, but only does when using the nightly build:

Unfortunately I cannot tell how to reproduce it, since I do not see the pattern. It just happens from time to time (a few times per day).

The touchscreen stops working 

(I can not scroll up/down using my fingers on the screen). But instead on the second monitor the programs start changing. Like when you hit shift tab.

I made a screencast (blurred it). Bottom screen is my convertible laptop with touchscreen. Top screen is an external monitor. I only touch the touchscreen and try to move the page up and down. This is what happens: 

https://user-images.githubusercontent.com/57603012/111316362-7c496080-8663-11eb-9825-0d043b0bda4b.mp4

Advice is very appreciated :)

**Comment by personalizedrefrigerator** [comment]
Do you have hand recognition turned on? If so, try disabling it.
![Internal hand recognition is under the touchscreen tab in settings.](https://user-images.githubusercontent.com/46334387/111347075-14cbea80-863c-11eb-8305-deaa34c936f7.png)

Internal hand recognition can mess with touchscreen settings, which might be causing this.

**Reply by KonstantinHebenstreit** [reply]
No, I do not have that turned on...
Do you have another idea what could cause it?

**Reply by KonstantinHebenstreit** [reply]
If I do enable internal hardware recognition, this specific problem is gone suprisingly! :)
But unfortunately it does not solve my problem, since then there is just normal hang up every few minutes, like a freeze. No touch possible, no writing with pen possible, and mouse cursor invisible (bug report #2920).

I think my problems are mostly caused by switching back and forth from Xournal to Firefox. I watch university lectures, which I pause very often, and write Notes in Xournal, to then continue with the Videos. It seems to my that Xournal does not 'like' this switching...

**Reply by personalizedrefrigerator** [reply]
# Determining what happens

This won't solve it, but it should be useful for debugging purposes.
In a terminal, run `xinput --list`.

For me, the output looks like this (on a device without a pen or touchscreen :) ):
```
WARNING: running xinput against an Xwayland server. See the xinput man page for details.
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ xwayland-pointer:17                     	id=6	[slave  pointer  (2)]
⎜   ↳ xwayland-relative-pointer:17            	id=7	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ xwayland-keyboard:17                    	id=8	[slave  keyboard (3)]
```

Find the id of your touchscreen in the output and run `xinput watch-props [the ID of your touchscreen here]`. According to `man xinput`, it should print to standard output when property changes occur.

# Things to try

* Try [switching from GNOME on X11 to GNOME on Wayland or from GNOME on Wayland to GNOME on X11.](https://linuxconfig.org/how-to-enable-disable-wayland-on-ubuntu-20-04-desktop)
* Clearing `xournalpp` preferences:
```sh
# Clearing preferences for v1.1.0
$ cd ~/.config
$ mv xournalpp xournalpp_backup

# Clearing preferences for v1.0.20
$ cd ~
$ mv .xournalpp .xournalpp_backup
$
```

**Reply by KonstantinHebenstreit** [reply]
Thanks for your ideas, so far I tried switching to X11 and also clearing my preferences (and of course the combination), but unfortunately nothing seemed to work...

I would like to save the output of my touchscreen, but to be honest I do not know on which id I should perform it, could you tell me?

Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ Dell Dell Universal Receiver Mouse      	id=10	[slave  pointer  (2)]
⎜   ↳ Dell Dell Universal Receiver Consumer Control	id=11	[slave  pointer  (2)]
⎜   ↳ ELAN0732:00 04F3:262B                   	id=15	[slave  pointer  (2)]
⎜   ↳ SynPS/2 Synaptics TouchPad              	id=18	[slave  pointer  (2)]
⎜   ↳ ELAN0732:00 04F3:262B Pen (0)           	id=22	[slave  pointer  (2)]
⎜   ↳ ELAN0732:00 04F3:262B Eraser (0)        	id=23	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ Power Button                            	id=6	[slave  keyboard (3)]
    ↳ Video Bus                               	id=7	[slave  keyboard (3)]
    ↳ Power Button                            	id=8	[slave  keyboard (3)]
    ↳ Dell Dell Universal Receiver            	id=9	[slave  keyboard (3)]
    ↳ Dell Dell Universal Receiver System Control	id=12	[slave  keyboard (3)]
    ↳ HP Wide Vision HD Camera: HP Wi         	id=13	[slave  keyboard (3)]
    ↳ HP IR Camera: HP IR Camera              	id=14	[slave  keyboard (3)]
    ↳ ELAN0732:00 04F3:262B                   	id=16	[slave  keyboard (3)]
    ↳ AT Translated Set 2 keyboard            	id=17	[slave  keyboard (3)]
    ↳ HP Wireless hotkeys                     	id=19	[slave  keyboard (3)]
    ↳ HP WMI hotkeys                          	id=20	[slave  keyboard (3)]
    ↳ Dell Dell Universal Receiver Consumer Control	id=21	[slave  keyboard (3)]

**Reply by KonstantinHebenstreit** [reply]
*I tried switching to wayland

**Reply by personalizedrefrigerator** [reply]
> ⎜ ↳ ELAN0732:00 04F3:262B id=15 [slave pointer (2)]

This looks like it!

**Comment by KonstantinHebenstreit** [comment]
Device 'ELAN0732:00 04F3:262B':
	Device Enabled (154):	1
	Coordinate Transformation Matrix (156):	1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000, 0.000000, 0.000000, 1.000000
	libinput Calibration Matrix (311):	1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000, 0.000000, 0.000000, 1.000000
	libinput Calibration Matrix Default (312):	1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000, 0.000000, 0.000000, 1.000000
	libinput Send Events Modes Available (276):	1, 0
	libinput Send Events Mode Enabled (277):	0, 0
	libinput Send Events Mode Enabled Default (278):	0, 0
	Device Node (279):	"/dev/input/event8"
	Device Product ID (280):	1267, 9771

**Comment by KonstantinHebenstreit** [comment]
This output of xinput watch-props stays the same, there are no lines added, when I use my notebook. Can I do anything differently to give you more info?

Btw. the error also occurs when working without a second screen.
